app.py

```diff
@@ -193,24 +266,52 @@ def main():
-    collateral_token_decimals = int(math.log10(src.settings.TOKEN_SETTINGS[collateral_token].decimal_factor))
-    underlying_addresses_to_decimals = {collateral_token_underlying_address: collateral_token_decimals}
-    prices = src.helpers.get_prices(token_decimals = underlying_addresses_to_decimals)
-    collateral_token_price = prices[collateral_token_underlying_address]
-    # TODO: Add Ekubo start
-    main_chart_data = main_chart_data.astype(float)
-    debt_token_underlying_address = src.helpers.UNDERLYING_SYMBOLS_TO_UNDERLYING_ADDRESSES[debt_token]
-    main_chart_data = add_ekubo_liquidity(
-        data=main_chart_data,
-        collateral_token=collateral_token_underlying_address,
-        debt_token=debt_token_underlying_address,
-    )
+
+    if current_pair == stable_coin_pair:
+        for stable_coin in src.settings.DEBT_TOKENS[:-1]:
+            debt_token = stable_coin
+            
+            # Fetch underlying addresses and decimals
+            collateral_token_underlying_address = src.helpers.UNDERLYING_SYMBOLS_TO_UNDERLYING_ADDRESSES[collateral_token]
+            collateral_token_decimals = int(math.log10(src.settings.TOKEN_SETTINGS[collateral_token].decimal_factor))
+            underlying_addresses_to_decimals = {collateral_token_underlying_address: collateral_token_decimals}
+
+            # Fetch prices
+            prices = src.helpers.get_prices(token_decimals=underlying_addresses_to_decimals)
+            collateral_token_price = prices[collateral_token_underlying_address]
+
+            # TODO: Add Ekubo start
+            main_chart_data = main_chart_data.astype(float)
+            debt_token_underlying_address = src.helpers.UNDERLYING_SYMBOLS_TO_UNDERLYING_ADDRESSES[debt_token]
+            main_chart_data = add_ekubo_liquidity(
+                data=main_chart_data,
+                collateral_token=collateral_token_underlying_address,
+                debt_token=debt_token_underlying_address,
+            )
+    else:
+        # Fetch underlying addresses and decimals
+        collateral_token_underlying_address = src.helpers.UNDERLYING_SYMBOLS_TO_UNDERLYING_ADDRESSES[collateral_token]
+        collateral_token_decimals = int(math.log10(src.settings.TOKEN_SETTINGS[collateral_token].decimal_factor))
+        underlying_addresses_to_decimals = {collateral_token_underlying_address: collateral_token_decimals}
+
+        # Fetch prices
+        prices = src.helpers.get_prices(token_decimals=underlying_addresses_to_decimals)
+        collateral_token_price = prices[collateral_token_underlying_address]
+
+        # TODO: Add Ekubo start
+        main_chart_data = main_chart_data.astype(float)
+        debt_token_underlying_address = src.helpers.UNDERLYING_SYMBOLS_TO_UNDERLYING_ADDRESSES[debt_token]
+        main_chart_data = add_ekubo_liquidity(
+            data=main_chart_data,
+            collateral_token=collateral_token_underlying_address,
+            debt_token=debt_token_underlying_address,
+        )
```

Comment:
Approach is correct! good job!

Nevertheless, code in lines 293-308 duplicates code in lines 275-290. To avoid duplication i suggest to move repeated logic into separate function. That would make code more readable.

Reply:
done! ✅  thank you!!